### ISSUE TITLE
AxiosRequestConfig httpAgent and httpsAgent are any

### ISSUE THREAD

**Issue by CSirolliTownleyIT** [axios/axios#6431]
I've noticed that in the AxiosRequestConfig type:
https://github.com/axios/axios/blob/9588fcdec8aca45c3ba2f7968988a5d03f23168c/index.d.ts#L183
both httpAgent and httpsAgent are set to `any`.
https://github.com/axios/axios/blob/9588fcdec8aca45c3ba2f7968988a5d03f23168c/index.d.ts#L210-L211

Shouldn't these be set to `http.Agent` and `https.Agent`?

**Comment by RopoMen** [comment]
Perhaps, but in browser side agent could be something else or can it be even set?

If you need http library only for Node.js checkout https://github.com/nodejs/undici

**Comment by CSirolliTownleyIT** [comment]
I'm fairly certain there is no agent that can be set in the browser. See [this StackOverflow question](https://stackoverflow.com/questions/66798599/node-https-agent-in-browser).

**Comment by jasonsaayman** [comment]
yeah cannot be set in the browser... i dont think it would be wise to pin them to a type other than any
